pkg/util/tests/kubectl/kubectl.go

```diff
@@ -0,0 +1,343 @@
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return "", fmt.Errorf("error getting pods: %v, output: %s", err, output)
+	}
+
+	return output, nil
+}
+
+// CreateNamespace creates a namespace
+// If the namespace already exists, it will return nil
+func CreateNamespace(ns string) error {
+	cmd := kubectl("create namespace %s", ns)
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		if strings.Contains(output, "AlreadyExists") {
+			return nil
+		}
+
+		return fmt.Errorf("error creating namespace: %v, output: %s", err, output)
+	}
+
+	return nil
+}
+
+// DeleteNamespace deletes a namespace
+func DeleteNamespace(ns string) error {
+	cmd := kubectl("delete namespace %s", ns)
+	_, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("error deleting namespace: %v", err)
+	}
+
+	return nil
+}
+
+// CheckNamespaceExist checks if a namespace exists
+// If the namespace exists, it will return nil
+// If the namespace does not exist, it will return an error
+func CheckNamespaceExist(ns string) error {
+	cmd := kubectl("get namespace %s -o jsonpath={metadata.name}", ns)
+	_, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			return ErrNotFound
+		}
+
+		return fmt.Errorf("error checking namespace: %v", err)
+	}
+
+	return nil
+}
+
+// GetDeployments returns the deployments of a namespace
+func GetDeployments(ns string) ([]string, error) {
+	cmd := kubectl("get deployments %s -o name", nsflag(ns))
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return nil, fmt.Errorf("error getting deployments names: %v, output: %s", err, output)
+	}
+	return extractNames(output), nil
+}
+
+// Delete deletes a resource based on the namespace, kind and the name
+func Delete(ns, kind, name string) error {
+	cmd := kubectl("delete %s %s %s", kind, name, nsflag(ns))
+	_, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return fmt.Errorf("error deleting deployment: %v", err)
+	}
+
+	return nil
+}
+
+// Logs returns the logs of a deployment
+// Arguments:
+// - ns: namespace
+// - selector: selector of the pod
+// - Since: time range
+func Logs(ns, selector string, since time.Duration) (string, error) {
+	cmd := kubectl("logs -l %s %s  --since=%s", selector, nsflag(ns), since)
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return "", err
+	}
+	return output, nil
+}
+
+// GetDaemonSets returns the daemonsets of a namespace
+// Return a list of daemonsets
+func GetDaemonSets(ns string) ([]string, error) {
+	cmd := kubectl("get daemonsets %s -o name", nsflag(ns))
+	output, err := shell.ExecuteCommand(cmd)
+	if err != nil {
+		return nil, fmt.Errorf("error getting daemonsets names: %v, output: %s", err, output)
+	}
+	return extractNames(output), nil
+}
+
+// DaemonSetPodsAllAvailable returns true if all the pods of the daemonset are available
+func DaemonSetPodsAllAvailable(ns string, daemonsetName string) error {
```

Comment:
```suggestion
func AllDaemonSetPodsAvailable(ns string, name string) error {
```

Reply:
I'm not a fan of functions like this. Too specific, not very reusable, and not very useful when the test fails. 

It would be much better to have an assertion like `Expect(numberAvailable).To(Equal(numberScheduled)` instead of just calling a function that performs this assertion internally and returns an error. 

If you still want this to be a function, then the function should be named `AssertAllDaemonSetPodsAvailable` and it should call `Expect`.

Reply:
Let me move this and make some changes to not hide any of the check